homeassistant/components/whirlpool/climate.py

```diff
@@ -107,16 +105,33 @@ class AirConEntity(ClimateEntity):
 
     def __init__(self, hass, said, name, backend_selector: BackendSelector, auth: Auth):
```

Comment:
There seem to be a few different changes in the PR..
- Adding an associated device and changing to use `has_entity_name`  w/ the device name
- Changing the callback and write state behavior
- Bumping the library version
- Adding the new sensors
And I think this should be broken down into separate smaller PRs that we can review quickly following the suggestions in https://developers.home-assistant.io/docs/creating_component_code_review#5-make-your-pull-request-as-small-as-possible  

Can you pre-factor some of those out into separate PRs?

Reply:
I know.... this thing has been growing since August when the first PR of the bunch became hung up. 
I'll start breaking it up.